Source/OxyPlot.Wpf/Series/Series.cs

```diff
@@ -64,7 +64,7 @@ public abstract class Series : ItemsControl
         /// </summary>
         static Series()
         {
-            VisibilityProperty.OverrideMetadata(typeof(Series), new PropertyMetadata(Visibility.Visible, AppearanceChanged));
+            VisibilityProperty.OverrideMetadata(typeof(Series), new PropertyMetadata(Visibility.Visible, DataChanged));
```

Comment:
Small example added. See Examples/WPF/LiveDemo

Steps to reproduces:
1. Enable both checkboxes (show and generate) for a few seconds. You should see data plotting.
2. Uncheck both
3. Toggle 'show' and verfiy data is still there
4. Check 'generate' and wait a few seconds before unchecking
5. Toggle 'show' to check if data is correct.

Prior the fix, the old data would still be shown.

Hope all it's clear.